# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Создаем сводную таблицу **children_debt_ratio** для подсчета колличества значений наличия задолженности у клиента в зависимости от колличества детей:
- *children* - колличество детей (от 0 до 5);
- *debt* - имел ли клиент задолженность по возврату кредита (0 - не имел, 1 - имел).


In [22]:
children_debt_ratio = data.pivot_table(index = 'children', 
                                       columns = 'debt', 
                                       values = 'days_employed', 
                                       aggfunc = 'count',
                                       margins = True) 
#в значении values считает общее колличество строк, поэтому можно выбирать любой столбец
children_debt_ratio['ratio_0'] = round(children_debt_ratio.loc[:,0] / children_debt_ratio.loc[:,'All']*100, 2)
children_debt_ratio['ratio_1'] = round(children_debt_ratio.loc[:,1] / children_debt_ratio.loc[:,'All']*100, 2)
children_debt_ratio.rename(columns = {0:'Без задолженности',
                                      1:'Просроченные кредиты',
                                      'All':'Всего займов',
                                      'ratio_0':'Без задолженности, %',
                                      'ratio_1':'Просроченные кредиты, %'})
#для удобства прочтения таблица была переименована

debt      Без задолженности  Просроченные кредиты  Всего займов  \
children                                                          
0                   13028.0                1063.0         14091   
1                    4364.0                 444.0          4808   
2                    1858.0                 194.0          2052   
3                     303.0                  27.0           330   
4                      37.0                   4.0            41   
5                       9.0                   NaN             9   
All                 19599.0                1732.0         21331   

debt      Без задолженности, %  Просроченные кредиты, %  
children                                                 
0                        92.46                     7.54  
1                        90.77                     9.23  
2                        90.55                     9.45  
3                        91.82                     8.18  
4                        90.24                     9.76  
5                       100.00                      NaN  
All                      91.88                     8.12

Процент просроченных кредитов и погашенных без задолженности кредитов был вычислен по формуле:
$$ Проценты = \frac{Кредит\:с/без\:задолженности}{Общее\:колличество\:выданных\:кредитов} $$

 **Вывод:** 

Из таблицы мы видим, что процент просроченных кредитов у клиентов без детей несколько ниже, чем у клиентов с детьми. Тем не менее, наличие 1 или более детей показывает приблезительно одинаковый процент выплат с задолженностью. Таким образом, можно разделить клиентов на две категории: без детей и с детьми, колличество же детей на процент выплат в срок не влияет. 

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Создаем сводную таблицу **family_debt_ratio** для подсчета колличества значений наличия задолженности у клиента в зависимости от семейного положения клиента:
- *family_status* - идентификатор семейного положения;
- *debt* - имел ли клиент задолженность по возврату кредита (0 - не имел, 1 - имел).

In [23]:
family_debt_ratio = data.pivot_table(index = 'family_status', 
                                       columns = 'debt', 
                                       values = 'days_employed', 
                                       aggfunc = 'count',
                                       margins = True) 
#в значении values считает общее колличество строк, поэтому можно выбирать любой столбец
family_debt_ratio['ratio_0'] = round(family_debt_ratio.loc[:,0] / family_debt_ratio.loc[:,'All']*100, 2)
family_debt_ratio['ratio_1'] = round(family_debt_ratio.loc[:,1] / family_debt_ratio.loc[:,'All']*100, 2)
family_debt_ratio.rename(columns = {0:'Без задолженности',
                                    1:'Просроченные кредиты',
                                    'All':'Всего займов',
                                    'ratio_0':'Без задолженности, %',
                                    'ratio_1':'Просроченные кредиты, %'})
#для удобства прочтения таблица была переименована

debt                   Без задолженности  Просроченные кредиты  Всего займов  \
family_status                                                                  
Не женат / не замужем               2523                   273          2796   
в разводе                           1105                    84          1189   
вдовец / вдова                       888                    63           951   
гражданский брак                    3749                   385          4134   
женат / замужем                    11334                   927         12261   
All                                19599                  1732         21331   

debt                   Без задолженности, %  Просроченные кредиты, %  
family_status                                                         
Не женат / не замужем                 90.24                     9.76  
в разводе                             92.94                     7.06  
вдовец / вдова                        93.38                     6.62  
гражданский брак                      90.69                     9.31  
женат / замужем                       92.44                     7.56  
All                                   91.88                     8.12

**Вывод:** 

Из таблицы видно, что клиенты, которые не женаты, либо живут в гражданском браке чаще склонны к просрочке кредита. А клиенты в браке, в разводе, или вдовы/вдовцы чаще платят без задолженностей. 

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Создаем сводную таблицу **income_debt_ratio** для подсчета колличества значений наличия задолженности у клиента в зависимости от уровня дохода клиента:
- *total_income_category* - категории по доходу, где  A - самая высокая,  E - самая низкая;
- *debt* - имел ли клиент задолженность по возврату кредита (0 - не имел, 1 - имел).

In [24]:
income_debt_ratio = data.pivot_table(index = 'total_income_category', 
                                       columns = 'debt', 
                                       values = 'days_employed', 
                                       aggfunc = 'count',
                                       margins = True) 
#в значении values считает общее колличество строк, поэтому можно выбирать любой столбец
income_debt_ratio['ratio_0'] = round(income_debt_ratio.loc[:,0] / income_debt_ratio.loc[:,'All']*100, 2)
income_debt_ratio['ratio_1'] = round(income_debt_ratio.loc[:,1] / income_debt_ratio.loc[:,'All']*100, 2)
income_debt_ratio.rename(columns = {0:'Без задолженности',
                                    1:'Просроченные кредиты',
                                    'All':'Всего займов',
                                    'ratio_0':'Без задолженности, %',
                                    'ratio_1':'Просроченные кредиты, %'})
#для удобства прочтения таблица была переименована

debt                   Без задолженности  Просроченные кредиты  Всего займов  \
total_income_category                                                          
A                                     23                     2            25   
B                                   4660                   354          5014   
C                                  14568                  1353         15921   
D                                    328                    21           349   
E                                     20                     2            22   
All                                19599                  1732         21331   

debt                   Без задолженности, %  Просроченные кредиты, %  
total_income_category                                                 
A                                     92.00                     8.00  
B                                     92.94                     7.06  
C                                     91.50                     8.50  
D                                     93.98                     6.02  
E                                     90.91                     9.09  
All                                   91.88                     8.12

**Вывод:** 

Из таблицы видно, что клиентов категории А и категории Е довольно мало для оценки их платёжеспособности, поэтому, по ним трудно сделать определенные выводы. Основные группы для статистики будут: 
- категория В, где процент просроченных кредитов ниже среднего, т.е. группа наиболее платежеспособная;
- категория С, где процент просроченных кредитов (без учета группы Е) будет самым высоким, т,е, вырастает риск просрочки по кредиту;
- категория D, где процент по просрочке самый низкий, но из за малого колличесва могут быть погрешности в расчетах. 

Таким образом, установлена зависимость между категорией уровня дохода клиента и возвратом кредита в срок. 

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Создаем сводную таблицу **purpose_debt_ratio** для подсчета колличества значений наличия задолженности у клиента в зависимости от категории цели получения кредита:
- *purpose_category* - категория целей получения кредита;
- *debt* - имел ли клиент задолженность по возврату кредита (0 - не имел, 1 - имел).

In [25]:
purpose_debt_ratio = data.pivot_table(index = 'purpose_category', 
                                       columns = 'debt', 
                                       values = 'days_employed', 
                                       aggfunc = 'count',
                                       margins = True) 
#в значении values считает общее колличество строк, поэтому можно выбирать любой столбец
purpose_debt_ratio['ratio_0'] = round(purpose_debt_ratio.loc[:,0] / purpose_debt_ratio.loc[:,'All']*100, 2)
purpose_debt_ratio['ratio_1'] = round(purpose_debt_ratio.loc[:,1] / purpose_debt_ratio.loc[:,'All']*100, 2)
purpose_debt_ratio.rename(columns = {0:'Без задолженности',
                                      1:'Просроченные кредиты',
                                      'All':'Всего займов',
                                      'ratio_0':'Без задолженности, %',
                                      'ratio_1':'Просроченные кредиты, %'})
#для удобства прочтения таблица была переименована

debt                      Без задолженности  Просроченные кредиты  \
purpose_category                                                    
операции с автомобилем                 3879                   400   
операции с недвижимостью               9971                   780   
получение образования                  3619                   369   
проведение свадьбы                     2130                   183   
All                                   19599                  1732   

debt                      Всего займов  Без задолженности, %  \
purpose_category                                               
операции с автомобилем            4279                 90.65   
операции с недвижимостью         10751                 92.74   
получение образования             3988                 90.75   
проведение свадьбы                2313                 92.09   
All                              21331                 91.88   

debt                      Просроченные кредиты, %  
purpose_category                                   
операции с автомобилем                       9.35  
операции с недвижимостью                     7.26  
получение образования                        9.25  
проведение свадьбы                           7.91  
All                                          8.12

**Вывод:** 

Из таблицы видно, что наиболее просроченные проценты по кредитам, у клиентов, которые берут займ для операций с автомобилем или получением образования. Клиенты, которые берут займ на операции с нежвижимостью или проведением свадьбы, чаще отдают кредит без задолженностей.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:*  в данной таблице основные проблемы с данными были в столбцах **days_employed**, где были отрицательные дни трудового стажа, и **total_income**, где были пропущены значения дохода клиентов. Данные ошибки могли быть вызваны либо по причине человеческого фактора, либо сбоем системы. 

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* математическим свойством медианы является то, что сумма абсолютных (по модулю) отклонений от медианного значения дает минимально возможное значение, если сравнивать с отклонениями от любой другой величины. То есть она более устойчева к аномальным значениям в данных.

### Шаг 4: общий вывод.

Проектная работа заключалась в обработке данных от кредитного отдела банка, для выяснения зависимстей наличия и колличества детей, семейного положения, уровня дохода и причин взятия кредита на факт погашения кредита в срок. 

В ходе выполнения проектной работы были выполнены действия по предобработке данных, в которых: были заменены пропуски в исходных данных по ежемесячному доходу клиентов (столбец **total_income**) на медианное значение для каждого типа заработка (столбец **income_type**). Данное значение позволило учесть особенности обработки данных без изменения статистических показателей. Также, были обработаны аномальные значения трудового стажа (столбец **days_employed**) и значения колличества детей (столбец **children**), которые не отражают действительности и появились по какой-либо ошибке (*человеческий фактор*, *програмная ошибка*). Помимо этого, были убраны дубликанты, приведены все данные к нижнему регистру. Заключительной частью предварительной обработки данных было изменение типов данных на целочисление в столбце ежемесячного дохода. Таким образом, все действия по предварительной обработке были выполнены. 

Далее, все клиенты были разбиты на категории по уровню дохода: 
   - С доходом выше 1000001 - категория 'А';
   - С доходом от 200001 до 1000000 - категория 'В';
   - С доходом от 50001 до 200000 - категория 'C';
   - С доходом от 30001 до 50000 - категория 'D';
   - С доходом до 30000 - категория 'E'.
   
Также, цели получения кредита были обобщены по следующим операциям:
   - Операции с автомобилем;
   - Операции с недвижимостью;
   - Проведение свадьбы;
   - Получение образования.
   
Исходя из предложеных категорий были выявлены зависимости, которые позволяют ответить на первоначальные вопросы:

1. Есть ли зависимость между колличеством детей и возвратом кредита в срок?

Да, статистика показывает, что у клиентов без детей просроченных кредитов **7.54%**, что меньше, чем средние показатели для клиентов с детьми **8.12%**. Более того, при выявлении статистики, стало видно, что при большем колличестве детей, растет процент по просроченным кредитам, таким образом, для клиентов с 4 детьми, процент просроченных кредитов уже достигает **9.76%**, что заметно выше среднего значения. 

2. Есть ли зависимость между семейным положением и возвратом кредита в срок?

По статистике, заемщики, которые ранее состояли или состоящие в браке реже допускают просрочку **7.06%** и **7.56%** соответственное. А для остальных категорий, процент просрочки сильно выше, для не женатых/не замужем он достигает **9.76%**, для состоящих в гражданском браке **9.31%**. Стоит учитывать, что самым низким показателем просроченных кредитов, обладает категория вдова/вдовец **6.62%**.

3. Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Было установлено, что при увеличении уровня дохода заемщики реже допускают задолженности по кредитам, так для категории ***Е*** процент задолженности по займу доходит до **9.09%**, для категории ***D*** составлет **6.02%**, для категории ***С*** будет **6.02%**, категории ***В*** будет **7.06%**, а для категории ***А*** уже **8%** ровно. Тем не менее, стоит также учесть, что статистических данных для категории ***А*** и категории ***Е*** довольно мало, чтобы сделать однозначные выводы по ним, поэтому они могут быть статистически незначительными. 

4. Как разные цели кредита влияют на его возврат в срок?

Из данных установлено, что наиболее обширной и добросовестной категорией платильщиков являются заемщики, которые берут кредит на операции с недвижимостью, для них процент просроченных кредитов состовляет **7.26%**, также, небольшой процент просроченных кредитов у заемщиков, чьей целбю получения кредита стало процедение свадьбы **7.91%**, а вот займы, выданные на получение образования и операции с автомобилем имеют наибольший процент просроченных кредитов **9.25%** и **9.35%** соответственно. 

Таким образом, были проведены работы по предварительной обработке и обработке данных, и получены необходимые зависимости. 